map/src/reader.rs

```diff
@@ -331,18 +331,20 @@ impl LayerTilemap {
         use format::ColorComponent::*;
         use format::LayerTilemapError::*;
 
-        fn extra<TS>(raw: &[i32], version: i32, flags: u32, too_short: TS)
+        fn extra<TS>(raw: &[i32], version: i32, flags: i32, too_short: TS)
             -> Result<&format::MapItemLayerV1TilemapExtraRace, format::LayerTilemapError>
             where TS: FnOnce(usize) -> format::LayerTilemapError
         {
             format::MapItemLayerV1TilemapExtraRace::from_slice(raw, version, flags)
-                .ok_or_else(|| too_short(raw.len()))
+                .map_err(|e| match e {
+                    TooShort(_) => too_short(raw.len()),
+                    other_err => other_err,
+                })
         }
 
         let v0 = format::MapItemLayerV1CommonV0::mandatory(raw, TooShort, InvalidVersion)?;
         let v2 = format::MapItemLayerV1TilemapV2::mandatory(raw, TooShortV2, InvalidVersion)?;
         let v3 = format::MapItemLayerV1TilemapV3::optional(raw, TooShortV3)?;
-        let flags = v2.flags as u32;
```

Comment:
v2.flags are i32, InvalidFlags() value is i32, but TILELAYERFLAG_* are u32.

I've decided to change a few functions to accept flags as i32 to avoid converting i32 -> u32 -> i32. This has removed the need for a separate u32 variable.

Reply:
The idea was that unsigned integers behave more like bit containers than signed integers (see arithmetic right shift of signed integers). The conversions and separate variables only exist in source code, to the CPU and the compiler, they live in the same location.

I'm fine with the change.